/*
 * This file was automatically generated by EvoSuite
 * Fri Nov 18 08:17:13 GMT 2022
 */

package org.math.array.util;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;
import org.math.array.util.Sorting;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class Sorting_ESTest extends Sorting_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      double[] doubleArray0 = new double[6];
      Sorting sorting0 = new Sorting(doubleArray0, false);
      sorting0.getIndex(0);
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      double[] doubleArray0 = new double[0];
      Sorting sorting0 = new Sorting(doubleArray0, false);
      sorting0.getIndex();
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      // Undeclared exception!
      try { 
        Sorting.invertIndex((int[]) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.math.array.util.Sorting", e);
      }
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      int[] intArray0 = new int[2];
      intArray0[0] = (-3234);
      // Undeclared exception!
      try { 
        Sorting.invertIndex(intArray0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // Index -3234 out of bounds for length 2
         //
         verifyException("org.math.array.util.Sorting", e);
      }
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      double[] doubleArray0 = new double[2];
      Sorting sorting0 = new Sorting(doubleArray0, false);
      // Undeclared exception!
      try { 
        sorting0.getIndex(2311);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // Index 2311 out of bounds for length 2
         //
         verifyException("org.math.array.util.Sorting", e);
      }
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      Sorting sorting0 = null;
      try {
        sorting0 = new Sorting((double[]) null, false);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.math.array.util.Sorting", e);
      }
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      double[] doubleArray0 = new double[7];
      doubleArray0[3] = (-2423.676221191469);
      Sorting sorting0 = new Sorting(doubleArray0, true);
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      double[] doubleArray0 = new double[1];
      Sorting sorting0 = new Sorting(doubleArray0, false);
      sorting0.getIndex(0);
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      int[] intArray0 = new int[0];
      Sorting.invertIndex(intArray0);
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      double[] doubleArray0 = new double[1];
      Sorting sorting0 = new Sorting(doubleArray0, false);
      int[] intArray0 = sorting0.getIndex();
      Sorting.invertIndex(intArray0);
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      Sorting.main((String[]) null);
  }
}
